x-pack/plugins/triggers_actions_ui/public/application/components/add_message_variables.tsx

```diff
@@ -34,39 +41,41 @@ export const AddMessageVariables: React.FunctionComponent<Props> = ({
   onSelectEventHandler,
   showButtonTitle = false,
 }) => {
+  const [isShowAllPressed, setIsShowAllPressed] = useState(false);
   const [isVariablesPopoverOpen, setIsVariablesPopoverOpen] = useState<boolean>(false);
 
-  const getMessageVariables = () =>
-    messageVariables?.map((variable: ActionVariable, i: number) => (
-      <EuiContextMenuItem
-        key={variable.name}
-        data-test-subj={`variableMenuButton-${variable.name}`}
-        icon="empty"
-        disabled={variable.deprecated}
-        onClick={() => {
-          onSelectEventHandler(variable);
-          setIsVariablesPopoverOpen(false);
-        }}
-      >
-        <>
-          <EuiText size="m" data-test-subj={`variableMenuButton-${i}-templated-name`}>
-            {templateActionVariable(variable)}
-          </EuiText>
-          <EuiText size="m" color="subdued">
-            <div>{variable.description}</div>
-          </EuiText>
-        </>
-      </EuiContextMenuItem>
-    ));
+  const { euiTheme } = useEuiTheme();
+
+  const messageVariablesObject: Record<string, ActionVariable> | undefined =
```

Comment:
lets wrap this in a `useMemo`

Reply:
Fixed here: 267a82876bec6a453b040e8cdaf55be70e92cf2c